tests/monitoring/monitoring_service/test_handlers.py

```diff
@@ -134,3 +144,246 @@ def test_channel_closed_event_handler_leaves_existing_channel(
+    assert context.db.channels[0].state == ChannelState.OPENED
+
+    event2 = ReceiveChannelClosedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        closing_participant='B',
+        block_number=52,
+    )
+    channel_closed_event_handler(event2, context)
+
+    assert len(context.db.channels) == 1
+    assert context.db.channels[0].state == ChannelState.CLOSED
+
+    event3 = ReceiveMonitoringNewBalanceProofEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        reward_amount=1,
+        nonce=2,
+        ms_address='C',
+        raiden_node_address='B',
+        block_number=23,
+    )
+
+    channel = context.db.get_channel(event3.token_network_address, event3.channel_identifier)
+    assert channel.update_status is None
+
+    monitor_new_balance_proof_event_handler(event3, context)
+
+    assert len(context.db.channels) == 1
+    channel = context.db.get_channel(event3.token_network_address, event3.channel_identifier)
+    assert channel.update_status is not None
+    assert channel.update_status.nonce == 2
+    assert channel.update_status.update_sender_address == 'C'
+
+    event4 = ReceiveMonitoringNewBalanceProofEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        reward_amount=1,
+        nonce=5,
+        ms_address='D',
+        raiden_node_address='B',
+        block_number=23,
+    )
+
+    monitor_new_balance_proof_event_handler(event4, context)
+
+    assert len(context.db.channels) == 1
+    channel = context.db.get_channel(event3.token_network_address, event3.channel_identifier)
+    assert channel.update_status is not None
+    assert channel.update_status.nonce == 5
+    assert channel.update_status.update_sender_address == 'D'
+
+
+def test_action_monitoring_triggered_event_handler_ignores_other_event(
+    context: Context,
+):
+    event = Event()
+
+    with pytest.raises(AssertionError):
+        action_monitoring_triggered_event_handler(
+            event=event,
+            context=context,
+        )
+
+
+def test_action_monitoring_triggered_event_handler_does_not_trigger_monitor_call_when_nonce_to_small(
+    context: Context,
+):
+    event = ReceiveChannelOpenedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        participant1='A',
+        participant2='B',
+        settle_timeout=100,
+        block_number=42,
+    )
+
+    assert len(context.db.channels) == 0
+    channel_opened_event_handler(event, context)
+
+    assert len(context.db.channels) == 1
+    assert context.db.channels[0].state == ChannelState.OPENED
+
+    event2 = ReceiveChannelClosedEvent(
+        token_network_address='abc',
+        channel_identifier=3,
+        closing_participant='B',
+        block_number=52,
+    )
+    channel_closed_event_handler(event2, context)
+
+    assert len(context.db.channels) == 1
+    assert context.db.channels[0].state == ChannelState.CLOSED
```

Comment:
Is everything up to this point the same as in the other tests? If yes, we can put it into a shared function or fixture. It's a bit much code to see the actually relevant part of each test, otherwise.

Reply:
True, I needed some external motivation to do this, copy and paste is so nice ;)